pkg/sql/sqlliveness/slstorage/slstorage.go

```diff
@@ -558,6 +578,38 @@ func (s *Storage) Update(
 		return false, hlc.Timestamp{}, errors.Wrapf(err, "could not update session %s", sid)
 	}
 	s.metrics.WriteSuccesses.Inc(1)
+	// If we were able to write to the sqlliveness table, then region communication
+	// may be restored. At this point lets attempt to clean up any unavailable
+	// region liveness time if it was set.
+	if resetUnavailableAtTime {
+		if err := s.txn(ctx, func(ctx context.Context, txn *kv.Txn) error {
+
+			// Check if this region has an unavailable_at time set.
+			unavailableRegions, err := s.livenessProber.QueryUnavailablePhysicalRegions(ctx, txn, false /*filterAvailable*/)
+			if err != nil {
+				return err
+			}
+			readTS := txn.ReadTimestamp()
+			// If it took us longer then the expiration to recover, then fail
+			// at this point. This could happen if the deadline is hit while
+			// recovering and we are forced to retry.
+			if expiration.Less(txn.ReadTimestamp()) {
```

Comment:
I think this operation doesn't actually care about the expiration timestamp. It only cares about the unavailable at instance. As long as the unavailable at instance is still in the future it can clear the row.

That also makes it easier to extract the logic into a helper in the region liveness package because using the unavailable_at instant as the deadline allows the reset logic to only cares about the state of the region liveness table.